lib/chef/provider/package/windows.rb

```diff
@@ -80,6 +93,63 @@ def install_package(name, version)
         def remove_package(name, version)
           package_provider.remove_package(name, version)
         end
+
+        # @return [Array] new_version(s) as an array
+        def new_version_array
+          # Because the one in the parent caches things
+          [new_resource.version]
+        end
+
+        private
+
+        def should_download?
+          is_url?(new_resource.source) && !::File.exists?(source_location)
+        end
+
+        def resource_for_provider
+          @resource_for_provider = Chef::Resource::WindowsPackage.new(new_resource.name).tap do |r|
+            r.source(Chef::Util::PathHelper.validate_path(source_location))
+            r.timeout(new_resource.timeout)
+            r.returns(new_resource.returns)
+            r.options(new_resource.options)
+          end
+        end
+
+        def download_source_file
+          source_resource.run_action(:create)
+          Chef::Log.debug("#{@new_resource} fetched source file to #{source_resource.path}")
+        end
+
+        def source_resource
```

Comment:
Yep, that shouldn't be hard to do. However, as I went down this road, it sort of felt like if you wanted to do anything fancy like that, you should just use remote file. Not sure if that is a valid argument as I do feel like I'm a little biased here.

I originally wanted to let people pass a remote file resource as the source attribute (with all the chef dsl-ing), however at that point, you may as well just declare it with the dsl that's already there.